pkg/apis/v1/ec2nodeclass_conversion_test.go

```diff
@@ -0,0 +1,538 @@
+			}
+		})
+		It("should convert v1 ec2nodeclass associate public ip address ", func() {
+			v1EC2NodeClass.Spec.AssociatePublicIPAddress = lo.ToPtr(true)
+			Expect(v1beta1EC2NodeClass.Spec.AssociatePublicIPAddress).To(BeNil())
+			Expect(v1EC2NodeClass.ConvertTo(ctx, v1beta1EC2NodeClass)).To(BeNil())
+			Expect(lo.FromPtr(v1beta1EC2NodeClass.Spec.AssociatePublicIPAddress)).To(BeTrue())
+		})
+		It("should convert v1 ec2nodeclass ami family", func() {
+			v1EC2NodeClass.Spec.AMIFamily = &AMIFamilyUbuntu
+			Expect(v1beta1EC2NodeClass.Spec.AMIFamily).To(BeNil())
+			Expect(v1EC2NodeClass.ConvertTo(ctx, v1beta1EC2NodeClass)).To(BeNil())
+			Expect(lo.FromPtr(v1beta1EC2NodeClass.Spec.AMIFamily)).To(Equal(v1beta1.AMIFamilyUbuntu))
+		})
+		It("should convert v1 ec2nodeclass user data", func() {
+			v1EC2NodeClass.Spec.UserData = lo.ToPtr("test user data")
+			Expect(v1beta1EC2NodeClass.Spec.UserData).To(BeNil())
+			Expect(v1EC2NodeClass.ConvertTo(ctx, v1beta1EC2NodeClass)).To(BeNil())
+			Expect(lo.FromPtr(v1beta1EC2NodeClass.Spec.UserData)).To(Equal(lo.FromPtr(v1EC2NodeClass.Spec.UserData)))
+		})
+		It("should convert v1 ec2nodeclass role", func() {
+			v1EC2NodeClass.Spec.Role = "test-role"
+			Expect(v1beta1EC2NodeClass.Spec.Role).To(Equal(""))
+			Expect(v1EC2NodeClass.ConvertTo(ctx, v1beta1EC2NodeClass)).To(BeNil())
+			Expect(v1beta1EC2NodeClass.Spec.Role).To(Equal(v1EC2NodeClass.Spec.Role))
+		})
+		It("should convert v1 ec2nodeclass instance profile", func() {
+			v1EC2NodeClass.Spec.InstanceProfile = lo.ToPtr("test-instance-profile")
+			Expect(v1beta1EC2NodeClass.Spec.InstanceProfile).To(BeNil())
+			Expect(v1EC2NodeClass.ConvertTo(ctx, v1beta1EC2NodeClass)).To(BeNil())
+			Expect(lo.FromPtr(v1beta1EC2NodeClass.Spec.InstanceProfile)).To(Equal(lo.FromPtr(v1EC2NodeClass.Spec.InstanceProfile)))
+		})
+		It("should convert v1 ec2nodeclass tags", func() {
+			v1EC2NodeClass.Spec.Tags = map[string]string{
+				"test-key-tag-1": "test-value-tag-1",
+				"test-key-tag-2": "test-value-tag-2",
+			}
+			Expect(v1beta1EC2NodeClass.Spec.Tags).To(BeNil())
+			Expect(v1EC2NodeClass.ConvertTo(ctx, v1beta1EC2NodeClass)).To(BeNil())
+			Expect(v1beta1EC2NodeClass.Spec.Tags).To(Equal(v1EC2NodeClass.Spec.Tags))
+		})
+		It("should convert v1 ec2nodeclass block device mapping", func() {
+			v1EC2NodeClass.Spec.BlockDeviceMappings = []*BlockDeviceMapping{
+				{
+					EBS: &BlockDevice{
+						DeleteOnTermination: lo.ToPtr(true),
+						Encrypted:           lo.ToPtr(true),
+						IOPS:                lo.ToPtr(int64(45123)),
+						KMSKeyID:            lo.ToPtr("test-kms-id"),
+						SnapshotID:          lo.ToPtr("test-snapshot-id"),
+						Throughput:          lo.ToPtr(int64(4512433)),
+						VolumeSize:          lo.ToPtr(resource.MustParse("54G")),
+						VolumeType:          lo.ToPtr("test-type"),
+					},
+					DeviceName: lo.ToPtr("test-device"),
+					RootVolume: false,
```

Comment:
can you change it to true so that it doesn't default to the primitive default?
```suggestion
					RootVolume: true,
```